sdcm/utils/gce_utils.py

```diff
@@ -47,34 +47,58 @@
                       'gcp-local-ssd-latency'} | {os.environ.get('SCT_GCE_PROJECT', 'gcp-sct-project-1')}
 
 
-def append_zone(region: str) -> str:
-    dash_count = region.count("-")
-    if dash_count == 2:
-        # return as is if region already has availability zone in it
-        return region
-    if dash_count != 1:
-        raise Exception(f'Wrong region name: {region}')
+def random_zone(region: str) -> str:
     availability_zones = SUPPORTED_REGIONS.get(region, None)
     if not availability_zones:
         raise Exception(f'Unsupported region: {region}')
-    return f"{region}-{random.choice(availability_zones)}"
+    return f"{random.choice(availability_zones)}"
+
+
+def get_gce_compute_instances_client() -> tuple[compute_v1.InstancesClient, dict]:
+    info = KeyStore().get_gcp_credentials()
+    credentials = service_account.Credentials.from_service_account_info(info)
```

Comment:
this function takes few sec's to execute and is used multiple times, maybe cache `get_gcp_credentials()` method?